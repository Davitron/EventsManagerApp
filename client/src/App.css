.App {
  text-align: center; }

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px; }

#sidenav-overlay {
  z-index: 100 !important; }

.App-header {
  background-color: #26a69a !important;
  color: white;
  position: relative;
  z-index: 2; }

.App-main {
  background-position: center center;
  background-size: cover;
  background-image: url("./assests/image/banner4.jpg");
  height: 100%;
  margin: 0 auto;
  /*left: 0;*/
  min-height: 100%;
  min-width: 100%;
  position: absolute;
  top: 0;
  width: 100%;
  overflow-x: hidden;
  overflow-y: hidden; }

.landing {
  margin-top: 200px !important; }

.signup {
  margin-top: 100px !important; }

.App-main h3 {
  font-weight: 900px; }

.App-main .App-overlay {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: #000;
  opacity: 0.3;
  filter: alpha(opacity=30);
  z-index: 1; }

.App-title {
  font-size: 1.5em; }

.App-intro {
  font-size: large; }

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg); }
  to {
    transform: rotate(360deg); } }

.App-signup {
  display: inline-block;
  padding: 32px 48px 0px 48px;
  border: 1px solid #EEE; }

.App-signup main {
  flex: 1 0 auto;
  margin-top: auto; }

.center-modal {
  display: flex; }

.center-modal .modal {
  position: relative;
  width: auto; }
