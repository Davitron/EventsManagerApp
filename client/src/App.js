import React, { Component } from 'react';
import Landing from './app/components/Landing/Landing';
import './App.css';

/**
 *
 */
class App extends Component {
  /**
   *@returns {*} view for langding page
   */
  render() {
    return (
      <div>
        <Landing />
      </div>
      // <div className="App">
      //   <header className="App-header">
      //     <img src={logo} className="App-logo" alt="logo" />
      //     <h1 className="App-title">Welcome to React</h1>
      //   </header>
      //   <p className="App-intro">
      //     To get started, edit <code>src/App.js</code> and save to reload.
      //   </p>
      // </div>
      
    );
  }
}

export default App;
