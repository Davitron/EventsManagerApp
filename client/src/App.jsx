import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { Router, Route, Switch } from 'react-router';
import Home from './app/components/Landing/Home/Home';
import SignIn from './app/components/Landing/SignIn/SignIn';
import SignUp from './app/components/Landing/SignUp/SignUp';
import ForgotPassword from './app/components/Landing/ForgotPassword/ForgotPassword';
import VerifyEmail from './app/components/Landing/VerifyEmail/VerifyEmail';
import VerifiedEmail from './app/components/Landing/VerifyEmail/Verified';
import PrivateRoute from './helpers/PrivateRoute';
import AuthChecker from './helpers/authChecker';
import history from '../src/helpers/history';
import configureStore from './store/configureStore';
import Landing from './app/components/Landing/Landing';
import './App.css';

const store = configureStore;

/**
 *
 */
class App extends Component {
  /**
 *@returns {routes}
 get all sub routes for this view
 */
  getChildRoutes() {
    ReactDOM.render(
      <Provider store={store}>
        <Router history={history}>
          <Switch>
            <Route exact path="/" component={Home} />
            <Route exact path="/SignIn" component={SignIn} />
            <Route exact path="/SignUp" component={SignUp} />
            <Route exact path="/ForgotPassword" component={ForgotPassword} />
            <Route exact path="/Users/Verify" component={VerifyEmail} />
            <PrivateRoute exact authenticated={AuthChecker.checkUserAuth()} path="/Users/Verified" component={VerifiedEmail} />
          </Switch>
        </Router>
      </Provider>,
      document.getElementById('landing')
    );
  }

  /**
   *@returns {*} view for langding page
   */
  render() {
    return (
      <div>
        <Landing />
      </div>
    );
  }
}

export default App;
