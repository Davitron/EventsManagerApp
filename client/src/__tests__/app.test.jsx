import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import App from '../components/App';

let mountedComponent;
let wrapper;

/**
 * @description Initialise the component
 *
 * @returns {object} mountedComponent - Mounte
 */
const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<App />);
  }
  return mountedComponent;
};

describe('Event Component', () => {
  beforeAll(() => {
    wrapper = getComponent();
  });


  it('component successfully rendered', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });
});
