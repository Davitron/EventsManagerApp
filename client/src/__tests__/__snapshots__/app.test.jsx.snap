// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component component successfully rendered 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <div>
        <Connect(Header) />
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/register"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/login"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/forgot-password"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/reset-password"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/verify"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/verified"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/centers/:centerId"
          />
          <Route
            component={[Function]}
            exact={true}
            path="/centers"
          />
          <UserRoute
            component={[Function]}
            exact={true}
            path="/events"
            redirectPath="/login"
          />
          <UserRoute
            component={[Function]}
            exact={true}
            path="/pending-events"
            redirectPath="/login"
          />
          <Route
            component={[Function]}
          />
        </Switch>
      </div>,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/register"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/forgot-password"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/reset-password"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/verify"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/verified"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/centers/:centerId"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/centers"
            />
            <UserRoute
              component={[Function]}
              exact={true}
              path="/events"
              redirectPath="/login"
            />
            <UserRoute
              component={[Function]}
              exact={true}
              path="/pending-events"
              redirectPath="/login"
            />
            <Route
              component={[Function]}
            />
          </Switch>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/register"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/forgot-password"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/reset-password"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/verify"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/verified"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/centers/:centerId"
              />,
              <Route
                component={[Function]}
                exact={true}
                path="/centers"
              />,
              <UserRoute
                component={[Function]}
                exact={true}
                path="/events"
                redirectPath="/login"
              />,
              <UserRoute
                component={[Function]}
                exact={true}
                path="/pending-events"
                redirectPath="/login"
              />,
              <Route
                component={[Function]}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/register",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/forgot-password",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/reset-password",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/verify",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/verified",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/centers/:centerId",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/centers",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/events",
                "redirectPath": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "component": [Function],
                "exact": true,
                "path": "/pending-events",
                "redirectPath": "/login",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "component": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <div>
          <Connect(Header) />
          <Switch>
            <Route
              component={[Function]}
              exact={true}
              path="/"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/register"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/login"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/forgot-password"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/reset-password"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/verify"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/verified"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/centers/:centerId"
            />
            <Route
              component={[Function]}
              exact={true}
              path="/centers"
            />
            <UserRoute
              component={[Function]}
              exact={true}
              path="/events"
              redirectPath="/login"
            />
            <UserRoute
              component={[Function]}
              exact={true}
              path="/pending-events"
              redirectPath="/login"
            />
            <Route
              component={[Function]}
            />
          </Switch>
        </div>,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "blank",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(Header) />,
            <Switch>
              <Route
                component={[Function]}
                exact={true}
                path="/"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/register"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/login"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/forgot-password"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/reset-password"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/verify"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/verified"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/centers/:centerId"
              />
              <Route
                component={[Function]}
                exact={true}
                path="/centers"
              />
              <UserRoute
                component={[Function]}
                exact={true}
                path="/events"
                redirectPath="/login"
              />
              <UserRoute
                component={[Function]}
                exact={true}
                path="/pending-events"
                redirectPath="/login"
              />
              <Route
                component={[Function]}
              />
            </Switch>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Route
                  component={[Function]}
                  exact={true}
                  path="/"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/register"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/login"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/forgot-password"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/reset-password"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/verify"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/verified"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/centers/:centerId"
                />,
                <Route
                  component={[Function]}
                  exact={true}
                  path="/centers"
                />,
                <UserRoute
                  component={[Function]}
                  exact={true}
                  path="/events"
                  redirectPath="/login"
                />,
                <UserRoute
                  component={[Function]}
                  exact={true}
                  path="/pending-events"
                  redirectPath="/login"
                />,
                <Route
                  component={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/register",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/forgot-password",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/reset-password",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/verify",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/verified",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/centers/:centerId",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/centers",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/events",
                  "redirectPath": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "component": [Function],
                  "exact": true,
                  "path": "/pending-events",
                  "redirectPath": "/login",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "component": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
